cmd/lifecycle/main.go

```diff
@@ -113,18 +113,20 @@ func (ch *DefaultCacheHandler) InitCache(cacheImageRef string, cacheDir string,
 }
 
 type DefaultRegistryHandler struct {
-	keychain authn.Keychain
+	keychain         authn.Keychain
+	insecureRegistry []string
 }
 
-func NewRegistryHandler(keychain authn.Keychain) *DefaultRegistryHandler {
+func NewRegistryHandler(keychain authn.Keychain, insecureRegistries []string) *DefaultRegistryHandler {
 	return &DefaultRegistryHandler{
-		keychain: keychain,
+		keychain:         keychain,
+		insecureRegistry: insecureRegistries,
 	}
 }
 
 func (rv *DefaultRegistryHandler) EnsureReadAccess(imageRefs ...string) error {
```

Comment:
This check makes difficult to test the `insecure-registry` feature properly so for now I'm skipping it.
We are doing this check in the analyzer only (and used also in the creator of course making the acceptance test difficult to create)